app/forms/otp_verification_form.rb

```diff
@@ -22,7 +22,7 @@ def valid_direct_otp_code?
   end
 
   def pattern_matching_otp_code_format
-    /\A\d{#{otp_code_length}}\Z/
+    /\A[a-z0-9]{#{otp_code_length}}\z/i
```

Comment:
TIL `\Z` matches before a newline, but `\z` includes it, so this will reject trailing newlines now, which is stricter but I think we want

Reply:
Do we need `[A-Za-z0-9]` ?

Reply:
no, because of the `/i` at the end there

Reply:
ah missed that